src/rm_topk.c

```diff
@@ -197,6 +197,9 @@ void TopKRdbSave(RedisModuleIO *io, void *obj) {
                                  topk->width * topk->depth * sizeof(Bucket));
     RedisModule_SaveStringBuffer(io, (const char *)topk->heap,
                                  topk->k * sizeof(HeapBucket));
+    for(uint32_t i = 0; i < topk->k; ++i) {
+        RedisModule_SaveStringBuffer(io, topk->heap[i].item, strlen(topk->heap[i].item) + 1);
```

Comment:
This assumes that the heap is full, and will crash when the items are NULL.

```
redis-cli TOPK.RESERVE test 3 5 4 0.9
redis-cli save
```
Will segfault here, since you're calling `strlen` on a NULL value.

Reply:
Fixed